File: benchmark/ruby/services/http_service.rb
require 'net/http'
require 'uri'

class HttpService
  def self.fetch_url(target_url)
    # Server-Side Request Forgery (SSRF) [CWE-918]
    uri = URI.parse(target_url)
    response = Net::HTTP.get_response(uri)
    response.body
  end
end

File: benchmark/csharp/Services/HttpService.cs
using System.Net.Http;
using System.Threading.Tasks;

namespace Benchmark.CSharp.Services
{
    public class HttpService
    {
        private static readonly HttpClient client = new HttpClient();

        public async Task<string> FetchResource(string url)
        {
            // Server-Side Request Forgery [CWE-918]
            HttpResponseMessage response = await client.GetAsync(url);
            return await response.Content.ReadAsStringAsync();
        }
    }
}
